### ISSUE TITLE
Font-locking in innermode breaks after typing something

### ISSUE THREAD

**Issue by juligreen** [polymode/polymode#251]
Initially, after entering a buffer, font locking in the innermode works perfectly (as can be seen in the first screenshot).

But after entering text in the inner chunks, font locking breaks for that chunk (see the other two screenshots).

This is my polymode config:
```
  (require 'polymode)
  (require 'js2-mode)

  (define-hostmode javascript-hostmode
    :mode 'js2-mode
    :protect-syntax t)
  (define-innermode lit-html-innermode
    :mode 'mhtml-mode
    :head-matcher "html[[:space:]]?`"
    :tail-matcher "`"
    :head-mode 'host
    :tail-mode 'host)
  (define-polymode polymer-mode
    :hostmode 'javascript-hostmode
    :innermodes '(lit-html-innermode))
  (add-to-list 'auto-mode-alist '("\\.js\\'" . polymer-mode))

```

My whole init.el can be found [here](https://github.com/juligreen/Configfiles/blob/master/.emacs.d/init.el)

![image](https://user-images.githubusercontent.com/25890405/67788107-f9296980-fa71-11e9-8940-daf8025b544f.png)

![image](https://user-images.githubusercontent.com/25890405/67788143-0c3c3980-fa72-11e9-93b7-9f9a4f372b1e.png)

![image](https://user-images.githubusercontent.com/25890405/67788152-19592880-fa72-11e9-8b80-e043da585f2e.png)

**Comment by vspinu** [comment]
Sorry for coming late on this. HTML is notoriously complex and is not designed to work as a inner mode. I was pretty successful with making the [polymode/poly-erb](https://github.com/polymode/poly-erb) work, but it's the other way around. I would like to make a full featured web mode available (probably based on the `web-mode.el`) at some point, but it's a non-trivial endeavor. 

Do you happen to know a working emacs mode that would reasonably well do what you want to do (aka js + html chunks)?

**Comment by jsmestad** [comment]
@vspinu could you elaborate a bit about what causes this and what would need to be done to try and fix it? I'd be happy to roll up my sleeves and try to make the necessary changes if you could give me some direction 🤞

**Comment by vspinu** [comment]
It's rather complex. I don't know all the details, but the core issue is that html mode font-lock and indentation are very complex and assume that the entire buffer is all html. It's kinda hard to get around that. But last time I looked 2 year ago. Things might have changed. Unfortunately to make it work one would need to be knowledgeable about both polymode and html mode at a pretty deep details.

**Comment by vspinu** [comment]
@jsmestad I am looking at your blog post. Did you mean helping with the web-mode? If so that would be great. I can quickly create a new `poly-web-mode` project and we can maybe put together a web-mode polymode. WDYT?

**Comment by jsmestad** [comment]
@vspinu yeah that'd be great. I think it's a long requested feature for JSX, EEx, ERB, etc users where things are mixed content.

BTW are there any performance things to be aware of with Polymode? (or mmm-mode for that matter)

**Comment by vspinu** [comment]
Polymode is generally very fast (at least it should be) as there are no copying between buffers at all. Cannot comment about mmm mode. 

> I think it's a long requested feature for JSX, EEx, ERB, etc users where things are mixed content.

What's the need more concretely? I am not a web developer. Basically, how do you see poly and web-mode interacting? In your elixir example you use web-mode for inner-html mode font-locking and indentation. Ok, that's straightforward. But would it make sense for poly-web-mode to piggyback on web-mode for indentation and (optionally) font-locking but to bring the inner modes in front (aka with proper emacs modes for chunks)?

**Comment by mrjbj** [comment]
would love it if someone could solve this problem or point me to a solution. Thanks for getting us as far as you have regardless, though.
